### PR TITLE
Adjust kvdb background threads

### PR DESCRIPTION
**Pull request by yiwu-arbug**
Signed-off-by: Yi Wu <yiwu@pingcap.com>

###  What have you changed?
Adjusting KVDB's max_background_jobs from 6 to 8 and max_subcompaction from 1 to 3. Currently with max_background_jobs=8 there will be 8/4=2 threads delegated to flush to avoid flush being blocked (see https://github.com/pingcap/rocksdb/blob/d21d976a19de19e83c329704eb62598ed2a030db/db/db_impl_compaction_flush.cc#L1814). Increasing max_subcompaction will (hopefully) speedup large L0->L1 compaction to avoid write stall.

###  What is the type of the changes?
Improvement

###  How is the PR tested?
check if benchbot report any regression

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?
no

###  Does this PR affect `tidb-ansible`?
need to update config template accordingly.

### CODE REVIEW

File: etc/config-template.toml [hunk 1]
```diff
@@ -308,7 +308,7 @@
 
 ## Represents the maximum number of threads that will concurrently perform a sub-compaction job by
 ## breaking it into multiple, smaller ones running simultaneously.
-# max-sub-compactions = 1
+# max-sub-compactions = 3
```
Comment (yiwu-arbug) [inline etc/config-template.toml]: seems the template is already out of sync with the actual max_background_jobs default.

**Comment by siddontang** [comment]
8 by default is too big for a low-level configuration machine? e.g, 8c16g?

File: src/config.rs [hunk 2]
```diff
@@ -698,7 +698,7 @@ impl Default for DbConfig {
             wal_ttl_seconds: 0,
             wal_size_limit: ReadableSize::kb(0),
             max_total_wal_size: ReadableSize::gb(4),
-            max_background_jobs: 6,
+            max_background_jobs: 8,
```
Comment (BusyJay) [inline src/config.rs]: How about limiting it lower than the CPU core count - 1?

**Comment by yiwu-arbug** [comment]
> Maybe add the interpretation in `config-template`
> 
> ```
> max_flushes = std::max(1, max_background_jobs / 4);
> max_compactions = std::max(1, max_background_jobs - max_flushes);
> ```

done.

**Comment by yiwu-arbug** [comment]
test passed. PTAL? @zhangjinpeng1987 @BusyJay @siddontang

**Pull request merged**